primitives/tf_encrypted/primitives/sodium/cc/ops.cc

```diff
@@ -23,7 +41,15 @@ REGISTER_OP("SodiumEasyBoxSealDetached")
     .Input("sk_sender: uint8")
     .Output("ciphertext: uint8")
     .Output("mac: uint8")
-    .SetIsStateful();
+    .SetIsStateful()
+    .SetShapeFn([](shape_inference::InferenceContext* c){
+        shape_inference::ShapeHandle plaintext_shape = c->input(0);
+        shape_inference::ShapeHandle dtype_shape = c->MakeShape({sizeof(float)});
```

Comment:
`sizeof(float)}` is hard coded, it should be inferred based on plaintext dtype.

Reply:
pushed a fix